src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -918,9 +923,9 @@ internal class UserCancelledTrustException : Exception
         internal struct CheckCertificateStateResult
         {
             public bool Result { get; }
-            public string Message { get; }
```

Comment:
This seems to only be used to log, and the log only contains this message. If this really can be null, we should probably fix the log message to at least tell us something when CheckCertificateStateResult.Message is null.

https://github.com/dotnet/aspnetcore/blob/f98959f630f916d74ccdb705bd02ab34b86fcf85/src/Servers/Kestrel/Core/src/Internal/LoggerExtensions.cs#L43-L47

Reply:
I can only be null when result is false.

I've renamed the properties on this type. Result -> Success, Message -> FailureMesssage. Makes what is going on more obvious.